lotus-soup/manifest.toml

```diff
@@ -90,6 +90,37 @@ instances = { min = 1, max = 100, default = 5 }
   mining_mode = { type = "enum", default = "synchronized", options = ["synchronized", "natural"] }
 
 
+[[testcases]]
+name = "drand-outage"
+instances = { min = 1, max = 100, default = 5 }
+
+  [testcases.params]
+  clients = { type = "int", default = 0 }
+  miners = { type = "int", default = 3 }
+  balance = { type = "float", default = 1 }
+  sectors = { type = "int", default = 1 }
+  role = { type = "string" }
+  genesis_timestamp_offset = { type = "int", default = 0 }
+
+
+  random_beacon_type = { type = "enum", default = "local-drand", options = ["mock", "local-drand", "external-drand"] }
+
+  # Params relevant to drand nodes. drand nodes should have role="drand", and must all be
+  # in the same composition group. There must be at least threshold drand nodes.
+  # To get lotus nodes to actually use the drand nodes, you must set random_beacon_type="local-drand"
+  # for the lotus node groups.
+  drand_period = { type = "duration", default="30s" }
+  drand_catchup_period = { type = "duration", default="10s" }
+  drand_threshold = { type = "int", default = 2 }
+  drand_gossip_relay = { type = "bool", default = true }
+  drand_log_level = { type = "string", default="info" }
+  suspend_events = { type = "string", default="", desc = "a sequence of halt/resume/wait events separated by '->'" }
```

Comment:
can it be directly specified as a string array in toml rather than 2 parsing formats?

Reply:
Yeah, it could be, but I don't think it matters much at this stage.

Honestly I am not sure that I like the halt/resume sequence to be time based and not epoch based as a start (time in Lotus is almost always expressed as epochs).

In any case the issues with these tests in general (automation, maintain dependencies, etc.) are much bigger I think.